### DISCUSSION TITLE
NoSQLBench Discussions

### DISCUSSION THREAD

**Discussion by jshook** [nosqlbench/nosqlbench#241] [General]
## 👋 Welcome!

We’re using Discussions as a place to connect with other members of our community and discuss ideas and methods.

Please use discussions here to ask questions, share ideas, and engage with other community members around how to use and improve NoSQLBench. If you want to help us elaborate on big ideas💡before putting them into NoSQLBench, this is the best place to do it.

**Comment by aravind-nallan-yb** [comment]
Hi Shooky,

Long time, no see. Hope you're doing well.

Yugabyte has a YCQL API which is Cassandra compatible. I'm wondering with the API restructure if i can reuse adapters-cqld4 to just switch out the driver with our own fork of the datastax driver. Is adapter-cqld4 extensible in this way? If not I'd like your guidance on how to accomplish this.

Thanks!
Aravind

**Reply by jshook** [reply]
Hey Aravind! 'tis a small world after all. The nbr module provides everything in the core runtime sans the specific drivers. You can use the NBLIBDIR env var to load specific driver adapter implementations. Please try it out and let me know if this works for you? I'll try to get this on the new docs site pretty soon.

**Comment by aravind-nallan-yb** [comment]
Hey Shooky

Looking to test variable cyclerate over a period of time. Is that possible today? If not any future work planned in this area?

Thanks
Aravind

**Reply by jshook** [reply]
I know this is late, but I'm going to answer it anyway.

You can do that. It is built into the scripting layer. Essentially, any workload you run with NB can be captured in its native scripting form simply by calling --show-script. This dumps out the scenario control script which is created on users' behalf for trivial cases.

If you modify the `activities.<alias>.rate` parameter with a string matching what you would use with an activity parameter, it will apply the change immediately and rates will adapt. The scripting docs cover some of this.

The recipe for this is pretty straightforward:
1. `start` your activity instead of `run`ning it.
2. While it is running asynchronously to the main control script, just loop over a time delay and setting the variable above as you need.
3. Watch it go!

My apologies for not catching this when you asked. It's been a very busy year!
